### ISSUE TITLE
FileInfo somehow trims the path passed in when it should not

### ISSUE THREAD

**Issue by daxian-dbw** [dotnet/runtime#70524]
### Description

FileInfo somehow trims the path passed in when it should not. Then its `FullName` and `Extension` properties have the trailing spaces trimmed, but its `Name` property still has the trailing spaces kept. See the repro steps below.

### Reproduction Steps

1. Create the file `C:\arena\tmp\abc.txt`
2. Create a `FileInfo` object with the string `C:\arena\tmp\abc.txt    ` (note that there are 4 trailing spaces).
3. The `Exists` property of the `FileInfo` object returns `true`, while I think it should return false because the file "abc.txt    " doesn't exist.
4. The `FullName` and `Extension` property of the `FileInfo` object don't have the trailing spaces, but the `Name` property has the trailing spaces kept.

```powershell
## run in PowerShell 7.2 or 7.3 preview (using .NET 7.preview)
$f = [System.IO.FileInfo]::new("C:\arena\tmp\abc.txt    ")
$f.Exists
$f.FullName + "::"
$f.Extension + "::"
$f.Name + "::"
```

### Expected behavior

Either `$f.Exists` should be `false` because the string used to create the `FileInfo` object has trailing spaces and make it a different file, or `$f.Name` should not contain the trailing spaces.

### Actual behavior

`$f.Exists` is `true`, and `$f.Name` has the trailing spaces, but `$f.FullName` and `$f.Extension` do not.

![image](https://user-images.githubusercontent.com/127450/172954594-90bcf251-c904-4271-9913-08192c79055c.png)

### Regression?

No

### Configuration

.NET 7-preview.4
Windows 10

**Comment by huoyaoyuan** [comment]
Windows native API `CreateFile` trims file name, so the trimming is not wrong.
We need to explicitly decide or document about which properties does the trim.

**Comment by GSPP** [comment]
Is it a good idea to have `Name` disagree with `FullName`?

**Comment by daxian-dbw** [comment]
> Windows native API `CreateFile` trims file name, so the trimming is not wrong.

That's fine to have this by-design behavior. Then `Name` property should be trimmed too, to be consistent with the `FullName` and `Extension` properties.

**Comment by adamsitnik** [comment]
I agree that the behaviors should be consistent and aligned. I am afraid we won't have the time to fix it for .NET 7. @daxian-dbw please let me know if this is very important for you

**Comment by daxian-dbw** [comment]
It's not a high priority. The inconsistent behavior gets exposed through PowerShell and thus we got an issue opened from the community: https://github.com/PowerShell/PowerShell/issues/16922

**Comment by hangy** [comment]
> Windows native API `CreateFile` trims file name, so the trimming is not wrong. We need to explicitly decide or document about which properties does the trim.

Might be important to note that the trimming doesn't happen on *nix (eg. in the `mcr.microsoft.com/powershell:debian-buster-slim` container):
```pwsh
PS /> $f = [System.IO.FileInfo]::new("abc.txt    ")
PS /> $f.Exists
False
PS /> $f.FullName + "::"
/abc.txt    ::
PS /> $f.Extension + "::"
.txt    ::
PS /> $f.Name + "::"
abc.txt    ::
```

**Comment by ichbinsteffen** [comment]
@daxian-dbw @adamsitnik 
I have prepared a PR for this: 
https://github.com/dotnet/runtime/pull/112964 

It's still a draft, I'm currently reading through the CONTRIBUTING.md to see if there are more things to do, but if you like you could have a look at it already.
